lib/Transforms/Loops/SeaSCEVUtils.cc

```diff
@@ -0,0 +1,41 @@
+#include "llvm_seahorn/Loops/SeaSCEVUtils.h"
+
+#include "llvm/Analysis/ScalarEvolutionAliasAnalysis.h"
+
+namespace llvm_seahorn {
+
+bool seaSCEVContainsMul(const llvm::SCEV *Expr) {
```

Comment:
If I understood the paper correctly, anything but the basic CR expends into a multiplication. A basic CR is a triple `(base, +, stride)`. A chain is `(a1, +, a2, +, ..., an)` and I think it will be a multiplication of the individual parts. We should double check this and at least add a comment to address later whenever we run into this problem again.